### PR TITLE
Add test for #1796, failing to stub Array.prototype.sort

### PR DESCRIPTION
**Pull request by mroderick**
This PR shows that the issue described in #1796 has been fixed (in c2c78143ffd381418b9847e8518cadd0741dfddc)

Once this is merged, we should create a new release.

#### How to verify - mandatory
1. Check out this branch
2. `npm install`
3. `npm test`

#### Checklist for author

- [x] `npm run lint` passes
- [x] References to standard library functions are [cached](https://github.com/sinonjs/sinon/pull/1523).

### CODE REVIEW

File: test/issues/issues-test.js [hunk 1]
```diff
@@ -508,4 +508,12 @@ describe("issues", function () {
             sinon.assert.calledWith(spy, secondObj);
         });
     });
+
+    describe("#1796 - cannot stub Array.prototype.sort", function () {
+        it("it should not fail with RangeError", function () {
+            var stub = sinon.stub(Array.prototype, "sort");
+            [1, 2, 3].sort();
```
Comment (fearphage) [inline test/issues/issues-test.js]: All tests should have assertions. Assert that this doesn't throw.

File: test/issues/issues-test.js [hunk 2]
```diff
@@ -508,4 +508,12 @@ describe("issues", function () {
             sinon.assert.calledWith(spy, secondObj);
         });
     });
+
+    describe("#1796 - cannot stub Array.prototype.sort", function () {
+        it("it should not fail with RangeError", function () {
+            var stub = sinon.stub(Array.prototype, "sort");
```
Comment (fearphage) [inline test/issues/issues-test.js]: This stub needs to `.callThrough()` to trigger the `RangeError`, right?

File: test/issues/issues-test.js (hunk 1 above)
Comment (mroderick) [reply test/issues/issues-test.js]: Good point. I've added an extra commit. Once happy with this PR, I'll fold the commits together

File: test/issues/issues-test.js (hunk 2 above)
Comment (mroderick) [reply test/issues/issues-test.js]: Nope, `.callThrough()` is not necessary to trigger the `RangeError`.

File: test/issues/issues-test.js [hunk 3]
```diff
@@ -508,4 +508,14 @@ describe("issues", function () {
             sinon.assert.calledWith(spy, secondObj);
         });
     });
+
+    describe("#1796 - cannot stub Array.prototype.sort", function () {
+        it("it should not fail with RangeError", function () {
+            refute.exception(function () {
+                var stub = sinon.stub(Array.prototype, "sort");
+                [1, 2, 3].sort();
```
Comment (fearphage) [inline test/issues/issues-test.js]: This should be the only line that we are concerned with not throwing an exception.

Comment (mroderick) [reply test/issues/issues-test.js]: Correct, I've pushed a commit that fixes that

**Comment by mroderick** [comment]
Dang, I merged without squashing ...

**Pull request merged**